c10/util/variant.h

```diff
@@ -336,7 +336,7 @@ namespace c10 {
   struct in_place_type_t { explicit in_place_type_t() = default; };
 
 #ifdef MPARK_VARIABLE_TEMPLATES
-  constexpr variant_in_place_t in_place{};
+  constexpr variant_in_place_t variant_in_place{};
```

Comment:
What's going on here?

Reply:
It's for avoiding conflict with the same variable name in `c10/util/Optional.h`. I added a comment in the c10-specific change log for better clarity.

Reply:
We should just make `variant.h` use the `optional.h` definition. `in_place` is part of the standard.

Reply:
I moved the common definition for `in_place` into `c10/util/in_place.h`, so that both `variant.h` and `optional.h` can use it.